app/Util/Helpers.hs

```diff
@@ -3,10 +3,25 @@
     Description : Contains general-use helper functions.
 -}
 module Util.Helpers
-    (safeHead) where
+    (safeHead, returnImageData) where
+
+import qualified Data.ByteString as BS (readFile)
+import qualified Data.ByteString.Base64 as BEnc (encode)
+import Happstack.Server
+import System.Directory (removeFile)
 
 -- | Given a list and a default value, returns the head of the list, or the default value
 -- if the list is empty.
 safeHead :: a -> [a] -> a
 safeHead listHead [] = listHead
 safeHead _ (listHead:_) = listHead
+
+-- | Creates and converts an SVG file to an image file, deletes them both and
+-- returns the image data as a response.
+returnImageData :: String -> String -> IO Response
+returnImageData svgFilename imageFilename = do
+    imageData <- BS.readFile imageFilename
+    _ <- removeFile imageFilename
+    _ <- removeFile svgFilename
+    let encodedData = BEnc.encode imageData
+    return $ toResponse encodedData
```

Comment:
All files should end with a new line. Please configure your text editor to automatically do this (you'll need to look up how, depending on which editor you're using)